ZeekPluginDynamic.cmake

```diff
@@ -120,9 +120,9 @@ function (zeek_add_dynamic_plugin ns name)
         target_include_directories(${target_name} PRIVATE ${BinPAC_INCLUDE_DIR})
     endif ()
 
-    # Write the 'magic' __bro_plugin__ file. We can do that once during CMake
+    # Write the 'magic' __zeek_plugin__ file. We can do that once during CMake
     # invocation since it won't change.
-    file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/__bro_plugin__" "${full_name}\n")
+    file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/__zeek_plugin__" "${full_name}\n")
```

Comment:
Can we just change this without transition/deprecation?

For some reason Debian/Ubuntu package `zkg` and if we do this here, then a very recent Zeek installation from source (without a bundled zkg) using the distribution provided `zkg` will minimally fail to disable installed plugins.

We also break any potential custom tooling for toggling loading of plugins that people may have written.

Following line is in spicy-plugin, and a bit of a data point that this isn't a complete internal detail:
https://github.com/zeek/spicy-plugin/blob/a3e38215e7eb3ee733132c43cc3942379b038fc4/CMakeLists.txt#L240

Doing this somehow backward compatible will be a headache though, so would also approve.

Reply:
Ugh, I was unaware they packaged it separately. This is part of why we tell people to try to only use the bundled `zkg`, because it's guaranteed to have all of the features/changes we expect. The problem is that the isn't anything special about this file that would warn someone that they shouldn't be using it. We could potentially fix Zeek itself to warn if it finds it, I guess.

Reply:
FTR, spicy-plugin is part of Zeek proper as of 6.0.0, so this is a non-issue from that version on. For earlier versions changes here shouldn't make a difference unless the plan was to bump it back into zeek-5.x.

Reply:
Hrm. I'm not sure how to handle this. It's a matter of where to add a warning. CMake can just create both files for a while and Zeek itself can handle both files as well. Adding a warning to a newer version `zkg` is meaningless if it's externally packaged. I guess since the file gets installed by `zkg`, a warning in Zeek itself would make enough sense.

Reply:
> Zeek itself can handle both files as well.

It'd actually need to recognize `__bro_plugin__.disabled` as well to cover the "old zkg disabled plugin scenario" and not load the plugin even if `__zeek_plugin__` exists. 

Think it's more pragmatic to change to `__zeek_plugin__`, see if it upsets anyone and move on otherwise.